### PR TITLE
Upgrade Travis CI for Arm SVE

### PR DESCRIPTION
**Pull request by xrq-phys**
cf. comments in #424 , this PR is to:

- Include `armsve` kernels and the `armsve` configuration to Travis CI tests.

For this reason, the following changes are done in the Travis CI config:

- Upgraded linux ver. from `trusty` to `focal` because `armsve` kernels requires GCC 10+ to compile and QEmu 4.4+ to emulate.
- Changed several toolchains used by Travis CI because previous toolchain is no longer available on `focal`.

To-dos:

- ~Travis CI seems also supporting Arm nodes.
   Compared to x86_64+QEmu, maybe using Arm64+ArmIE a better way for testing Arm kernels?
   At least `armv8a` and `armv7a` can be tested without emulation.
   ArmIE's better stability over QEmu can be another advantage.~

### CODE REVIEW

**Comment by xrq-phys** [comment]
BTW, I'm not sure about benefits but by switching from travis-ci.org to travis-ci.com we will be able to use `arm64-graviton2` in addition to `arm64`.

**Comment by devinamatthews** [comment]
@xrq-phys how did you make the PR a draft? I've never been able to find that button or whatever it is.

File: .travis.yml
Comment (devinamatthews) [inline .travis.yml:76]: Why add `CFLAGS=$CFLAGS`? This doesn't seem to get set to anything special.

**Review by devinamatthews — APPROVED** [review]
Looks good, just one comment below.

**Comment by devinamatthews** [comment]
I would leave the arm64 HW support for another PR. We would probably want to wait until after migrating to travis-ci.com for that anyways so graviton2 can be included.

File: .travis.yml
Comment (xrq-phys) [reply .travis.yml:76]: This is from the old `.travis.yml`.
BLIS' `configure` is not Autoconf so I thought influence of args could be somehow different from env.

If it is not the case, I guess it's enough to just: `$DIST_PATH/configure -t $THR $CONF` ?

Comment (devinamatthews) [reply .travis.yml:76]: The old line 74 doesn't have CFLAGS (or at least, what I see here in the diff on the LHS doesn't). I would remove it for consistency.

Comment (xrq-phys) [reply .travis.yml:76]: Sorry.
The old `.travis.yml` has only `CC=$CC`.

I used `CFLAGS=-DSVE_XXX` upon other configs but now CFLAGS is not set anyway so this is an unnecessary extra by myself. Will remove.

**Comment by xrq-phys** [comment]
Got it.

I've set up travis.com build on my fork so hopefully native arm64 support (maybe ppc also?) will be there as soon as master's CI moves :)

**Comment by fgvanzee** [comment]
@xrq-phys We have credits ready to go for travis-ci.com, but we are delaying the migration for now because it requires us to click a button that say we are using the "beta" version of the service. (I asked their support about this, and they basically said, "Oh, don't worry about that. It's full service." I did not find their reply super convincing, although it's better than silence I suppose.)

**Comment by devinamatthews** [comment]
@fgvanzee this PR looks good to me. I'll merge if you're OK with it (note the travis.com-related stuff @xrq-phys mentioned is **not** in this PR).

**Comment by fgvanzee** [comment]
> (note the travis.com-related stuff @xrq-phys mentioned is not in this PR).

Understood.

It looks fine to me.

**Comment by fgvanzee** [comment]
Also, it appears that this PR obviates #453.

**Pull request merged**
